truth_set/java/structure/organisation/duplication/UserService.java

```diff
@@ -0,0 +1,122 @@
+        this.id = id;
+        this.name = name;
+        this.age = age;
+        this.isActive = true;
+    }
+
+    // Getters and setters
+    public String getId() { return id; }
+    public void setId(String id) { this.id = id; }
+    public String getName() { return name; }
+    public void setName(String name) { this.name = name; }
+    public int getAge() { return age; }
+    public void setAge(int age) { this.age = age; }
+    public boolean isActive() { return isActive; }
+    public void setActive(boolean active) { isActive = active; }
+    public String getEmail() { return email; }
+    public void setEmail(String email) { this.email = email; }
+    public String getPhone() { return phone; }
+    public void setPhone(String phone) { this.phone = phone; }
+    public Date getUpdatedAt() { return updatedAt; }
+    public void setUpdatedAt(Date updatedAt) { this.updatedAt = updatedAt; }
+    public boolean isHasEmailChanged() { return hasEmailChanged; }
+    public void setHasEmailChanged(boolean hasEmailChanged) { this.hasEmailChanged = hasEmailChanged; }
+    public boolean isHasPhoneChanged() { return hasPhoneChanged; }
+    public void setHasPhoneChanged(boolean hasPhoneChanged) { this.hasPhoneChanged = hasPhoneChanged; }
+}
+
+class UserService {
+    private Database database;
+
+    public UserService(Database database) {
+        this.database = database;
+    }
+
+    public User createUser(String name, int age) {
+        User user = new User(generateUniqueId(), name, age);
+        database.save(user);
+        return user;
+    }
+
+    public User addUser(String name, int age) {
+        User user = new User(generateUniqueId(), name, age);
+        database.save(user);
+        return user;
+    }
+
+    public void updateUserEmail(String userId, String email) {
+        User user = database.findUserById(userId);
+        if (user != null) {
+            user.setEmail(email);
+            user.setUpdatedAt(new Date());
+            user.setHasEmailChanged(true);
+            database.save(user);
+        }
+    }
+
+    public void updateUserPhone(String userId, String phone) {
+        User user = database.findUserById(userId);
+        if (user != null) {
+            user.setPhone(phone);
+            user.setUpdatedAt(new Date());
+            user.setHasPhoneChanged(true);
+            database.save(user);
+        }
+    }
+
+    public boolean isAdult(int age) {
+        return age >= 18;
+    }
+
+    public boolean canVote(int age) {
+        if (age >= 18) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean canDrink(int age) {
+        if (age >= 21) {
+            return true;
+        }
+        return false;
+    }
+
+    public double calculateDiscount(double price) {
+        double discountRate = 0.1;
+        return price * discountRate;
+    }
+
+    public double calculateTax(double price) {
+        double taxRate = 0.1;
+        return price * taxRate;
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider consolidating rate calculation methods</b></div>
<div id="fix">

The `calculateDiscount` and `calculateTax` methods have identical implementations with different variable names. Consider consolidating into a single rate calculation method. A similar issue was also found in truth_set/java/structure/organisation/duplication/UserService.java (line 49-59).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -99,12 +99,8 @@
 -    public double calculateDiscount(double price) {
 -        double discountRate = 0.1;
 -        return price * discountRate;
 -    }
 -
 -    public double calculateTax(double price) {
 -        double taxRate = 0.1;
 -        return price * taxRate;
 -    }
 +    public double calculateRate(double price, String rateType) {
 +        double rate = 0.1;
 +        return price * rate;
 +    }
```


```suggestion
    public double calculateRate(double price, String rateType) {
        double rate = 0.1;
        return price * rate;
    }
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/3#issuecomment-2522127787>#dcd59f</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged